src/constants.h

```diff
@@ -31,6 +31,17 @@ constexpr int RECENT_CONNECTIONS_MAX_COUNT = 5;
 // Cooldown period for unresponsive servers
 constexpr uint32_t SERVER_UNRESPONSIVE_COOLDOWN_SEC = 300;
 
+// Number of msecs for max runtime of the connection benchmarks.
+constexpr uint32_t BENCHMARK_MAX_DURATION_PING = 5000;
+constexpr uint32_t BENCHMARK_MAX_DURATION_DOWNLOAD = 15000;
+// TODO: Decide on thresholds for connection speeds
+constexpr uint32_t BENCHMARK_THRESHOLD_SPEED_FAST = 3125000;    // 25 Megabit
+constexpr uint32_t BENCHMARK_THRESHOLD_SPEED_MEDIUM = 1250000;  // 10 Megabit
+// TODO: Replace with self-hosted file
```

Comment:
If we're looking for a resource that's self-hosted, I think we can pull some static resources off of `archive.mozilla.org` which seems to be geographically well distributed thanks to Cloudlflare. This may require some testing and evaluation.

Reply:
That’s perfect — thanks!